homeassistant/components/madvr/__init__.py

```diff
@@ -0,0 +1,68 @@
+"""The madvr-envy integration."""
+
+from __future__ import annotations
+
+import logging
+
+from madvr.madvr import Madvr
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+
+from .coordinator import MadVRCoordinator
+
+PLATFORMS: list[Platform] = [Platform.REMOTE]
+
+type MadVRConfigEntry = ConfigEntry[MadVRCoordinator]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: MadVRConfigEntry) -> bool:
+    """Set up the integration from a config entry."""
+    madVRClient = Madvr(
+        host=entry.data[CONF_HOST],
+        logger=_LOGGER,
+        port=entry.data[CONF_PORT],
+        connect_timeout=10,
+        loop=hass.loop,
+    )
+    coordinator = MadVRCoordinator(hass, madVRClient)
+
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
Right, so the integration is pull based so this actually does nothing. Is there a way we can still confirm the device is present at that place and that we are able to succeed with setting up?

Reply:
>Is there a way we can still confirm the device is present at that place and that we are able to succeed with setting up?

only way to confirm the device is to connect to it, and it must be on to do so

Reply:
Where do we connect with it exactly?

Reply:
async_add_tasks starts the library background workers

once the library detects the device is pingable, it automatically initiates a connection (if its not connected already)

It also opens a connection in config flow after testing it and maintains that